libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.ts

```diff
@@ -284,15 +284,15 @@ export class ChartJSService {
     // However the stock chart, should have autogenerated labels if no
     // custom labels are supplied.
     const isStockType = type === 'stock';
-    const labels = !dataLabels
+    const labelsToApply = !labels
       ? isStockType
         ? this.getDefaultStockLabels(datasets, this.locale)
         : this.createBlankLabels(datasets)
-      : dataLabels;
+      : labels;
 
```

Comment:
A bit hard to read double ternary: Consider using if else and subsequently a single ternary expression.

Reply:
Modified the suggestion to rely on a `const` instead of `let` as i prefer to use immutable variables when i can get away with it. It has been changed to: 
```Typescript
    const labelsToApply = (() => {
      if (!labels && isStockType) return this.getDefaultStockLabels(datasets, this.locale);
      else if (!labels) return this.createBlankLabels(datasets);
      else return labels;
    })();
```

```